pandas/io/pytables.py

```diff
@@ -3196,6 +3206,11 @@ def create_index(self, columns=None, optlevel=None, kind=None):
 
                 # create the index
                 if not v.is_indexed:
+                    if v.type.startswith('complex'):
+                        raise TypeError('Columns containing complex values cannot be indexed when '
```

Comment:
This would be the case if ``data_columns=True``, e.g. all are indexed. So you can say that you should not use a complex column as a ``data_column``.